### PR TITLE
PUBDEV-3546: h2o.year() method does not return year

### PR DESCRIPTION
**Pull request by navdeep-G**
Ensure h2o.year() returns the year instead of years since 1900

---
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/372)

### CODE REVIEW

File: h2o-r/h2o-package/R/frame.R [hunk 1]
```diff
@@ -880,11 +880,12 @@ h2o.dct <- function(data, destination_frame, dimensions, inverse=FALSE) {
...
 #' This method calls the function of the MutableDateTime class in Java.
 #' @param x An H2OFrame object.
-#' @return An H2OFrame object containig the entries of \code{x} converted to years
-#'         starting from 1900, e.g. 69 corresponds to the year 1969.
+#' @return An H2OFrame object containing the entries of \code{x} converted to years
+#'         This calculation starts from 1900.
 #' @seealso \code{\link{h2o.month}}
 #' @export
-h2o.year <- function(x) .newExpr("-",.newExpr("year", chk.H2OFrame(x)),1900)
+h2o.year <- function(x) .newExpr("-",.newExpr("year", chk.H2OFrame(x)),1900) + 1900
```
Comment (jangorecki) [inline h2o-r/h2o-package/R/frame.R]: It does look like we subtract `1900` from year to later add `1900` to it. Maybe the following will be enough?
```r
.newExpr("year", chk.H2OFrame(x))
```
It make sense to wrap whole call into `as.integer` as we don't expect non-integer values.

File: h2o-r/h2o-package/R/frame.R [hunk 2]
```diff
@@ -880,11 +880,12 @@ h2o.dct <- function(data, destination_frame, dimensions, inverse=FALSE) {
 # is this still true?
 #' This method calls the function of the MutableDateTime class in Java.
 #' @param x An H2OFrame object.
-#' @return An H2OFrame object containig the entries of \code{x} converted to years
-#'         starting from 1900, e.g. 69 corresponds to the year 1969.
+#' @return An H2OFrame object containing the entries of \code{x} converted to years
+#'         This calculation starts from 1900.
```
Comment (jangorecki) [inline h2o-r/h2o-package/R/frame.R]: This is not the case already so this single line can be removed.

**Comment by navdeep-G** [comment]
I updated the tests, but I will look into making more for h2o primitives in a separate PR.

**Comment by navdeep-G** [comment]
@h2o-ops please test

**Comment by abal5** [comment]
@h2o-ops please test

**Comment by navdeep-G** [comment]
@h2o-ops please test

**Comment by navdeep-G** [comment]
@h2o-ops please test

**Comment by h2o-ops** [comment]
Tests passed!

**Pull request closed without merging**